frontend/model/captureLogs.js

```diff
@@ -0,0 +1,176 @@
+      localStorage.removeItem(`giConsole/${prevEntry}`)
+      prevEntry = entry.prev
+      entriesCount--
+      toDelete--
+    } while (toDelete && prevEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+    localStorage.setItem('giConsole/count', entriesCount)
+  }
+}
+
+export function captureLogsStart () {
+  console.debug('captureLogsStart called')
+  isCapturing = true
+  lastEntry = localStorage.getItem('giConsole/lastEntry')
+  entriesCount = +localStorage.getItem('giConsole/count') || 0
+
+  // Set a new visit or session - useful to understand logs through time.
+  // NEW_SESSION -> The user opened a new browser or tab.
+  // NEW_VISIT -> The user comes from an ongoing session (refresh or login)
+  const isNewSession = !sessionStorage.getItem('NEW_SESSION')
+  if (isNewSession) { sessionStorage.setItem('NEW_SESSION', 1) }
+  captureLog(isNewSession ? 'NEW_SESSION' : 'NEW_VISIT')
+
+  // Subscribe to appLogsFilter
+  const state = sbp('state/vuex/state')
+  appLogsFilter = state.appLogsFilter || []
+  sbp('okTurtles.events/on', SET_APP_LOGS_FILTER, filter => { appLogsFilter = filter })
+
+  console.debug('Starting to capture logs of type:', appLogsFilter)
+
+  // Override the console to start capturing the logs
+  if (!isConsoleOveridden) {
+    // avoid duplicated captures, in case captureLogsStart
+    // is called multiple times. (ex: login twice in the same visit)
+    isConsoleOveridden = true
+
+    // NOTE: Find a way to capture logs without messing up with log file location.
+    // console.log() doesnt include stack trace, so when logged, we can't access
+    // where the log came from (file name), which difficults debugging if needed.
+    window.console = new Proxy(window.console, {
+      get: (obj, type) => (...args) => {
+        obj[type](...args)
+        if (isCapturing && appLogsFilter.includes(type)) {
+          captureLog(type, ...args)
+        }
+      }
+    })
+  }
+}
+
+export function captureLogsPause () {
+  isCapturing = false
+  sbp('okTurtles.events/off', SET_APP_LOGS_FILTER)
+}
+
+// Util to download *all* stored logs so far
+export function downloadLogs (filename, elLink) {
+  const logsArr = []
+  let lastEntry = localStorage.getItem('giConsole/lastEntry')
+  do {
+    const entry = localStorage.getItem(`giConsole/${lastEntry}`)
+    const entryParsed = JSON.parse(entry) || {}
+    lastEntry = entryParsed.prev
+    logsArr.push(entry)
+  } while (lastEntry)
+  const data = `/*
+. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 
+
+GROUP INCOME - Application Logs
+
+Attach this file when reporting a problem:
+  - Github: https://github.com/okTurtles/group-income-simple/issues  
+
+. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
+*/
+
+var userAgent = ${navigator.userAgent}
+var appLogs = [${logsArr}]
+`
+
+  const file = new Blob([data], { type: 'text/plain' })
```

Comment:
I've changed my mind... we do not want people sending us random javascript files to execute. That could be dangerous.

So instead:

1. Save this as a `.json` file instead of `.txt`
2. Set the mimetype to `'application/json'`
3. Output JSON data by setting the `data` to `JSON.stringify(logsArr)`

Verify that it works by downloading the file and using node to load it back jusing `JSON.parse` and the nodejs fileReadSync stuff.

Reply:
>  we do not want people sending us random javascript files to execute. That could be dangerous.

I agree, but this is a .txt file, not a .js file. And when using a .json file, if we then need to use `JSON.parse` to read the file, it will still execute the code, right? So, we would have the same security problem.

Anyway... changed to JSON and it seems work fine (tested on an online JSON reader)

Reply:
The file extension shouldn't make much of a difference, it just tells the OS to treat the file as JSON.

> it will still execute the code, right?

No, JSON is JSON, not code.

Reply:
yes, but when the JSON is parsed it will be code. no?

Reply:
Nope.